test/performance/operf.go

```diff
@@ -0,0 +1,634 @@
+	err := c.CoreV1().PersistentVolumeClaims(*testNamespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, listPvcOptions)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func getPrometheusURL(c *kubernetes.Clientset) string {
+	data, err := c.RESTClient().
+		Get().
+		AbsPath("/apis/route.openshift.io/v1/namespaces/openshift-monitoring/routes/prometheus-k8s").
+		DoRaw(context.TODO())
+	if err != nil {
+		fmt.Println("Error getting Openshift Prometheus route:", err)
+		return ""
+	}
+	var route map[string]interface{}
+	err = json.Unmarshal([]byte(data), &route)
+	if err != nil {
+		panic("Failed to Unmarshal route")
+	}
+	host := route["spec"].(map[string]interface{})["host"]
+	return fmt.Sprintf("https://%s", host)
+}
+
+func (unit lvmUnitMetrics) getValues(metrics string) (float64, float64) {
+	/* Metrics have the following structure:
+	{"status":"success",
+	 "data":{"resultType":"matrix",
+	         "result":[
+	                    {"metric":{"pod":"topolvm-controller-58c8f86445-gxtb4"},
+	                     "values":[[1641983701,"164327424"],[1641983707,"164327424"],...]
+	                    }
+	                   ]
+	        }
+	}
+	*/
+	type subject struct {
+		Pod string
+	}
+	type series struct {
+		Metric subject
+		Values [][]interface{}
+	}
+	type resultData struct {
+		ResultType string
+		Result     []series
+	}
+	type metricsData struct {
+		Status string
+		Data   resultData
+	}
+	var d metricsData
+	var max float64 = 0
+	var avg float64 = 0
+	var n float64 = 0
+	err := json.Unmarshal([]byte(metrics), &d)
+	if err != nil {
+		panic("Failed to Unmarshal metrics")
+	}
+
+	if len(d.Data.Result) == 0 {
+		panic("Metrics Result is empty")
+	}
+	values := d.Data.Result[0].Values
+	for i := 0; i < len(values); i++ {
+		v, err := strconv.ParseFloat(fmt.Sprintf("%s", values[i][1]), 64)
+		if err != nil {
+			fmt.Println("error converting ", values[i][1], " to number")
+		} else {
+			if v > max {
+				max = v
+			}
+			avg += v
+			n++
+		}
+	}
+	if n > 0 {
+		avg = avg / n
+	} else {
+		avg = 0
+	}
+	return max, avg
+}
+
+func (unit lvmUnitMetrics) getMetrics(start int64, end int64, query string) string {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec
+	}
+	client := &http.Client{Transport: tr}
+
+	params := url.Values{}
+	params.Add("start", strconv.FormatInt(start, 10))
+	params.Add("end", strconv.FormatInt(end, 10))
+	params.Add("step", "6")
+	params.Add("namespace", "lvm-operator-system")
+	params.Add("query", query)
+	params.Add("timeout", "30s")
+	body := strings.NewReader("")
+
+	metricsQuery := fmt.Sprintf("%s/api/v1/query_range?%s", prometheusURL, params.Encode())
+	req, err := http.NewRequest("GET", metricsQuery, body)
+
+	if err != nil {
+		fmt.Printf("Error querying Prometheus: %s", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", *token))
```

Comment:
Hello @jmolmo Is there a way to use the kubeconfig itself to get the metrics and remove the use of token?

Reply:
Thanks @jmolmo